### DISCUSSION TITLE
[4.1.x][Proposal] CRUD routes - also support fqdn

### DISCUSSION THREAD

**Discussion by tabacitu** [Laravel-Backpack/community-forum#332] [Ideas]
For a while now, Freek from Spatie has been pushing for using
```diff
-    Route::get('monster', 'MonsterController');
+    Route::get('monster', /App/Http/Controllers/MonsterController);
```

Most recently in [this Laracasts episode](https://laracasts.com/series/guest-spotlight/episodes/5).

While I don't necessarily agree that this should be the _default_ for every Laravel project (because sometimes you don't use Resource Routes, but want to take the user to a certain method in that controller), I do like how it would fit _Backpack_ projects for routing to CRUDs. Since CRUD routes _are_ resource routes, you only need to point to the controller, not a certain method inside it.

So what I propose is that we:

**(1) Add support for this syntax:**

```diff
-    Route::crud('monster', 'MonsterCrudController');
+    Route::crud('monster', /App/Http/Controllers/Admin/MonsterCrudController::class);
```
(without removing the old one, of course)

**(2) Maybe make this one the default?** The one that's generated by our packages.

Thoughts anyone?

PS. This _would_ mean that we don't use the "namespace" feature in the route group. So yes, you'll have the ```/App/Http/Controllers/Admin``` namespace littered all across the ```routes/backpack/custom.php``` file. Basically for every CRUD route. But it _would_ have the benefit of being able to click the controller to go to it. And easy refactoring using IDEs. There's always a catch 😄 Not 100% it's worth it in this case. Looking for feedback.

**Comment by tabacitu** [comment]
See https://github.com/Laravel-Backpack/community-forum/issues/231

**Reply by karandatwani92** [reply]
I tested this on the latest version, and …
#### (1) support for this syntax is already working fine.
#### (2) But when I made the following default one inside(AddCustomRouteContent.php):
`/App/Http/Controllers/Admin/{nameTitle}CrudController::class)`

One has to be manually remove `namespace` attribute from `Route:group` and append namespace to other routes for once.
Then this would become compatible for future usage.

@tabacitu  What should we do?
